src/pydvl/valuation/scorers/classwise.py

```diff
@@ -96,12 +96,16 @@ def __init__(
         )
         self._in_class_discount_fn = in_class_discount_fn
         self._out_of_class_discount_fn = out_of_class_discount_fn
-        self.label = initial_label
+        self.label: int | None = None
         self.num_classes = len(np.unique(self.test_data.y))
 
     def __str__(self) -> str:
         return self.name
 
+    def with_label(self, label: int) -> Self:
```

Comment:
What is the reason to not expose the label parameter at the init anymore? Since the label is needed for this class to function properly, it is a bit unexpected to have to call `with_label` after the init.

Reply:
I have changed this to just be a property that no longer needs a method to be changed.